src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -233,6 +238,40 @@ private async Task RunLoop(CancellationToken cancellationToken)
                 }
             }
 
+            TimeSpan cancellationTime = TimeSpan.Parse(_options.Value.LogScrapingTimeout ?? "00:10:00");
+
+            try
+            {
+                _logger.LogInformation("Starting log scraping");
+
+                var logScrapingTimeoutCancellationTokenSource = new CancellationTokenSource(cancellationTime);
+                var logScrapingTimeoutCancellationToken = logScrapingTimeoutCancellationTokenSource.Token;
+
+                var combinedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, logScrapingTimeoutCancellationToken);
```

Comment:
I think it's really important to dispose this token source, otherwise they will keep accumulating inside the cancellationToken (every "linked" source means the original token need an event handler to tell the linked source it's time to fire).

Normal sources don't matter much, because they don't really have anything in them that needs disposing (you _can_ dispose them, it just doesn't matter). But these ones do.

Reply:
Added using.
I was not aware of this, thanks